pkg/tfbridge/provider.go

```diff
@@ -59,6 +59,96 @@ type Resource struct {
 	TFName string           // the Terraform resource name.
 }
 
+// runTerraformImporter runs the Terraform Importer defined on the Resource for the given
+// resource ID, and returns a replacement input map if any resources are matched.
+func (res *Resource) runTerraformImporter(resourceID resource.ID) (map[string]string, error) {
+	// There is nothing to do here if the resource doesn't have an importer defined in the
+	// Terraform schema.
+	if res.TF.Importer == nil {
+		return nil, nil
+	}
+
+	glog.V(9).Infof("%s has TF Importer", res.TFName)
+
+	id := resourceID.String()
+
+	// Prepare a Terraform ResourceData for the importer
+	data := res.TF.Data(nil)
+	data.SetId(id)
+	data.SetType(res.TFName)
+
+	// Run the importer defined in the Terraform resource schema
+	results, err := res.TF.Importer.State(data, nil)
+	if err != nil {
+		return nil, errors.Wrapf(err, "importing %s", id)
+	}
+
+	// No resources were returned - error out
+	if len(results) < 1 {
```

Comment:
Just out of curiosity, what happens if the importer is called for a resource that has been deleted out-of-band? Could it legitimately return 0 resources?

Reply:
Would get not fail in this case anyway during refresh?

Reply:
No: it would return a `nil` state, which the engine recognizes as “this resource was deleted.”

Reply:
Ah, then it is a possibility. What would we expect to happen if a resource we tried to get were deleted or never existed? Presumably continuing blind would just result in further errors?

Reply:
IIRC the engine or language provider will turn a missing resource during a `get` into an error.

Reply:
I've modified the importer code to respect this behaviour and use the engine behaviour for dealing with an empty state from import.